ouroboros-network/test/Test/Pipe.hs

```diff
@@ -94,43 +103,32 @@ demo chain0 updates = do
              | update <- updates
         ]
 
-    atomically $ takeTMVar consumerDone
+    atomically $ takeTMVar done
```

Comment:
I think that this test should wait on the producer and consumer asyncs (`waitBoth`) and only attempt to take the `done` TMVar if both asyncs complete successfully so that if the testcase fails it won't simply deadlock on `takeTMVar done`.